encoding/json/inbound_test.go

```diff
@@ -4,12 +4,12 @@ import (
 	"bytes"
 	"encoding/json"
 	"io"
-	"io/ioutil"
 	"reflect"
 	"testing"
 
 	"github.com/yarpc/yarpc-go"
 	"github.com/yarpc/yarpc-go/transport"
+	"github.com/yarpc/yarpc-go/transport/testing"
```

Comment:
we should look at what other packages do for their testing packages. e.g. std library uses suffixes:
`http` has `httptest`:
https://golang.org/pkg/net/http/httptest/

`io` has `iotest`:
https://golang.org/pkg/testing/iotest/

I'd definitely avoid having a leaf path name that isn't the package name. At the root, it's trickier, and since you can't have "-" in the name, it isn't as confusing, but having `transport/testing` import a `test_transport` is confusing.

Take a look at other popular libraries to see what they do as well.